src/coffee-script.coffee

```diff
@@ -84,10 +96,17 @@ exports.compile = compile = withPrettyErrors (code, options) ->
     js = "// #{header}\n#{js}"
 
   if options.sourceMap
-    answer = {js}
-    answer.sourceMap = map
-    answer.v3SourceMap = map.generate(options, code)
-    answer
+    v3SourceMap = map.generate(options, code)
+
+    if options.inlineMap
+      sourceMapDataURI = "//# sourceMappingURL=data:application/json;base64,#{base64encode v3SourceMap}"
+      sourceURL        = "//# sourceURL=#{options.filename ? 'coffeescript'}"
+      "#{js}\n#{sourceMapDataURI}\n#{sourceURL}"
```

Comment:
Mind explaining to me why sourceURL is needed? Also, are you sure things work (do you test several browsers?) having _both_ at the same time?